flow/layers/physical_shape_layer_unittests.cc

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/flow/layers/physical_shape_layer.h"
+
+#include "gtest/gtest.h"
+
+namespace flow {
+
+TEST(PhysicalShapeLayer, TotalElevation) {
+  std::shared_ptr<PhysicalShapeLayer> layers[4];
```

Comment:
nit: I think this can be unique?

Reply:
Unfortunately `ContainerLayer::Add` takes a `std::shared_ptr` so I can't make a `unique_ptr` here. After the `Add`, both the parent layer and the `layers` array here would have a reference to the layer.